curriculum/challenges/english/07-scientific-computing-with-python/learn-lambda-functions-by-building-an-expense-tracker/666ab82c874add3c95145213.md

```diff
@@ -0,0 +1,77 @@
+---
+id: 666ab82c874add3c95145213
+title: Step 7
+challengeType: 20
+dashedName: step-7
+---
+
+# --description--
+
+One last important method, for now, is the `pop()` method. `pop()` can be used to remove an element from a list. By default, it removes the last element of the list, but you can pass an argument to the method and it will remove the element at the given index.
+
+```py
+fruits_list = ["cherry", "lemon", "tomato", "apple", "orange"]
+
+fruits_list.pop(2)
+
+print(fruits_list) # ["cherry", "lemon", "apple", "orange"]
+```
+
+In this case, `fruits_list.pop(2)` removes the element at index `2` of the list.
+
+Use `pop()` to remove the last element from `my_list`, then print `my_list`.
+
+# --hints--
+
+You should use `pop()`.
+
+```js
+({
+    test: () => runPython(`
+    assert len(_Node(_code).find_calls('pop')) > 0
+    `)
+})
+```
+
+`my_list` should be `[0, 1, 2]`.
+
+```js
+({
+    test: () => runPython(`
+    assert my_list == [0, 1, 2]
+    `)
+})
+```
+
+You should print `my_list` after using `pop()`.
+
+```js
+({
+    test: () => runPython(`
+    assert _Node(_code).is_ordered('my_list.pop()', 'print(my_list)')
+    `)
+})
+```
```

Comment:
Not sure how, but it would be nice if we allowed the `pop()` to use an index argument here.

I haven't looked at the python testing enough, so all I have is a regex.

```js
({ test: () =>  {
    const commentless_code = __helpers.python.removeComments(code);
    assert.match(commentless_code, /my_list\.pop\(3?\)\s*print\(my_list\)\s*/);
  }
})
```